### DISCUSSION TITLE
deploying-asp-net-core-applications-to-kubernetes-part-8-running-database-migrations-using-jobs-and-init-containers/

### DISCUSSION THREAD

**Discussion by andrewlock** [andrewlock/blog-comments#341] [Announcements]
# deploying-asp-net-core-applications-to-kubernetes-part-8-running-database-migrations-using-jobs-and-init-containers/

In this post I show how to run database migrations using Kubernetes jobs and init containers

https://andrewlock.net/deploying-asp-net-core-applications-to-kubernetes-part-8-running-database-migrations-using-jobs-and-init-containers/

**Comment by cfra** [comment]
Thank you very much for this detailed blog post.

It provided me with good ideas and starting points for developing a strategy how to properly run migrations for our Django app in k8s. :+1:

This post really is applicable to more than just ASP.NET Core applications, its reasoning and decision making probably applies to a lot of applications that are using a relational database.

**Reply by akhanalcs** [reply]
Consider moving that Django app to .NET 8. We have cloud native framework called [Aspire](https://devblogs.microsoft.com/dotnet/introducing-dotnet-aspire-simplifying-cloud-native-development-with-dotnet-8/) in .NET 8. It's amazing!

[a comment by an automated account is not shown]

**Reply by cfra** [reply]
The migration job resource is created, runs once to completion and is maintained after that and tracks the completed state.

So, no matter when a new pod of the deployment is started, it will be able to query the (completed) status of the migration job.

Otherwise, this would also break pods that need to be restarted e.g. due to liveness check failures or node failures.

[a comment by an automated account is not shown]

**Comment by akhanalcs** [comment]
I faced the issue of init container failing. I fixed it by using this [solution](https://stackoverflow.com/a/77533386/8644294).
